webcomponents/tool/src/main/frontend/js/sakai-permissions.js

```diff
@@ -65,6 +65,16 @@ class SakaiPermissions extends SakaiElement {
   get tool() { return this._tool; }
 
   updated(changedProperties) {
+    let inputs = this.renderRoot.querySelectorAll("input[type='checkbox']");
+    inputs.forEach(elem => { 
+      let role = elem.getAttribute('data-role');
+      let perm = elem.getAttribute('data-perm');
+      
+      if(role && perm){
+        elem.checked = this.on[role].includes(perm);
+      }
+    })
```

Comment:
I'd move this code into your resetPermissions function body. You do this by waiting on the updateComplete promise.

this.updateComplete.then(() => {
let inputs = this.renderRoot.querySelectorAll("input[type='checkbox']");
    inputs.forEach(elem => { 
      let role = elem.getAttribute('data-role');
      let perm = elem.getAttribute('data-perm');

      if (role && perm) {
        elem.checked = this.on[role].includes(perm);
      }
    })
});

... you get the idea. That way the  render has finished and you can manipulate the checks.

Reply:
Nice! I like the idea. So, no need to reload or re-render. Sorry but, I'm still learning about webcomponents and litelement.

I think I've set all the blank lines and spaces as you like. Anyway, if anything else is needed, just say it.

Thanks for your help.

Reply:
The dispatchEvent part is to reset the table cell style arround the checkbox. 
We also can do this calling this.attachHandlers() or just:
`$('.permissions-table input:checkbox').change();`

But I don't like the idea to mix vanilla javascript with jquery (at least inside the same fragment/method)